dawn/prototype/AtlasFromNetcdf.cpp

```diff
@@ -0,0 +1,275 @@
+  auto ghost = atlas::array::make_view<int, 1>(nodes.ghost());
+  auto flags = atlas::array::make_view<int, 1>(nodes.flags());
+
+  auto radToLat = [](double rad) { return rad / (0.5 * M_PI) * 90; };
+  auto radToLon = [](double rad) { return rad / (M_PI)*180; };
+
+  for(int nodeIdx = 0; nodeIdx < numNodes; nodeIdx++) {
+    // following the same pattern here as in
+    lonlat(nodeIdx, atlas::LON) = radToLon(lon[nodeIdx]);
+    lonlat(nodeIdx, atlas::LAT) = radToLat(lat[nodeIdx]);
+
+    glb_idx_node(nodeIdx) = nodeIdx;
+    remote_idx(nodeIdx) = nodeIdx;
+
+    part(nodeIdx) = myPart;
+    ghost(nodeIdx) = false;
+    atlas::mesh::Nodes::Topology::reset(flags(nodeIdx));
+  }
+
+  return true;
+}
+
+bool CellsFromNetCDF(const netCDF::NcFile& dataFile, atlas::Mesh& mesh) {
+  auto [cellToVertex, vertexPerCell, ncells] = Load2DField<int>(dataFile, "vertex_of_cell");
+  if(vertexPerCell != 3) {
+    std::cout << "not a triangle mesh\n";
+    return false;
+  }
+
+  // define cells and associated properties
+  mesh.cells().add(new atlas::mesh::temporary::Triangle(), ncells);
+  auto cells_part = atlas::array::make_view<int, 1>(mesh.cells().partition());
+  atlas::mesh::HybridElements::Connectivity& node_connectivity = mesh.cells().node_connectivity();
+  atlas::array::ArrayView<atlas::gidx_t, 1> glb_idx_cell =
+      atlas::array::make_view<atlas::gidx_t, 1>(mesh.cells().global_index());
+
+  for(int cellIdx = 0; cellIdx < ncells; cellIdx++) {
+    // indices in netcdf are 1 based, data is column major
+    atlas::idx_t tri_nodes[3] = {cellToVertex[0 * ncells + cellIdx] - 1,
+                                 cellToVertex[1 * ncells + cellIdx] - 1,
+                                 cellToVertex[2 * ncells + cellIdx] - 1};
+    node_connectivity.set(cellIdx, tri_nodes);
+    glb_idx_cell[cellIdx] = cellIdx;
+    cells_part(cellIdx) = myPart;
+  }
+
+  return true;
+}
+
+template <typename ConnectivityT>
+bool AddNeighborList(const netCDF::NcFile& dataFile, atlas::Mesh& mesh,
+                     const std::string nbhListName, int yPerXExpected,
+                     ConnectivityT& connectivity) {
+  auto [xToY, yPerX, numY] = Load2DField<int>(dataFile, nbhListName);
+  if(yPerX != yPerXExpected) {
+    std::cout << "number of neighbors per element not as expected!\n";
+    return false;
+  }
+  for(int elemIdx = 0; elemIdx < numY; elemIdx++) {
+    atlas::idx_t yOfX[yPerX];
+    for(int innerIdx = 0; innerIdx < yPerX; innerIdx++) {
+      // indices in netcdf are 1 based, data is column major
+      yOfX[innerIdx] = xToY[innerIdx * numY + elemIdx] - 1;
+    }
+    connectivity.set(elemIdx, yOfX);
+  }
+  return true;
+}
+
+// Pre-allocate neighborhood table, snippet taken from
+// https://github.com/ecmwf/atlas/blob/98ff1f20dc883ba2dfd7658196622b9bc6d6fceb/src/atlas/mesh/actions/BuildEdges.cc#L73
+void AllocNbhTable(atlas::mesh::HybridElements::Connectivity& connectivity, int numElements,
```

Comment:
I missed it in the first review, but templating here also would remove the code duplication.